# MATH 014 (01) – Introduction to Data Science  
## Project 2 – Part 1  
**Department of Mathematics**  
**College of Arts and Sciences, Howard University**  
**Student Name:** Alexandra Zeppeiro  
**Student ID:** @03017718  
**Due Date:** 04-4-2025  

---

## Title: Mapping and Analyzing Black Population Trends Across U.S. States: A Demographic Study

### Introduction  
<span style="color:red; font-size:16px"><em>The "County-Level Demographic: Population, Race, Gender" dataset provides a comprehensive breakdown of demographic information for counties across the United States. Sourced from the U.S. Census Bureau's 2023 American Community Survey (ACS), this dataset includes detailed information on population counts by gender, race, and ethnicity, along with geographic identifiers for each county. The primary purpose of this dataset is to enable researchers, policymakers, and analysts to understand demographic distributions, track trends over time, and support geographic studies. It facilitates data-driven decision-making by providing insights for public policy, urban planning, and social research. The dataset is particularly valuable for demographic analysis, geographic studies, and creating visualizations to explore trends in population diversity across U.S. counties. Its public domain licensing ensures accessibility for a wide range of applications, making it a versatile tool for anyone interested in exploring demographic patterns at the county level.

### Dataset Source  
<span style="color:red; font-size:16px"><em>Dataset obtained from Kaggle: https://www.kaggle.com/datasets/ahmedmohamed2003/county-level-demographic-population-race-gender?resource=download

---

## SECTION I: DATASET OVERVIEW AND STRUCTURE (10 POINTS)


<span style="color:green; font-size:18px"><strong>1. What is the Shape and Structure of the Dataset?</strong></span><br>
- Use `.shape`, `.info()`, `.head()`, `.tail()`<br>
- Show first and last few rows of the dataset.<br>
- Describe the general information of the DataFrame.


In [946]:
import pandas
df = pandas.read_csv('demographic_data.csv')
df

County        State  State FIPS Code  County FIPS Code  \
0          Autauga County      Alabama                1                 1   
1          Baldwin County      Alabama                1                 3   
2          Barbour County      Alabama                1                 5   
3             Bibb County      Alabama                1                 7   
4           Blount County      Alabama                1                 9   
...                   ...          ...              ...               ...   
3217  Vega Baja Municipio  Puerto Rico               72               145   
3218    Vieques Municipio  Puerto Rico               72               147   
3219   Villalba Municipio  Puerto Rico               72               149   
3220    Yabucoa Municipio  Puerto Rico               72               151   
3221      Yauco Municipio  Puerto Rico               72               153   

       FIPS  Total Population  Male Population  Female Population  \
0      1001             59285            28669              30616   
1      1003            239945           117316             122629   
2      1005             24757            12906              11851   
3      1007             22152            11824              10328   
4      1009             59292            29934              29358   
...     ...               ...              ...                ...   
3217  72145             54058            25765              28293   
3218  72147              8147             4178               3969   
3219  72149             21778            10510              11268   
3220  72151             29868            14381              15487   
3221  72153             33509            15920              17589   

      Total Race Responses  White Alone  Black or African American Alone  \
0                    59285        43616                            11829   
1                   239945       198721                            19144   
2                    24757        10891                            11616   
3                    22152        16634                             4587   
4                    59292        53062                              747   
...                    ...          ...                              ...   
3217                 54058        13681                             2249   
3218                  8147         1028                              222   
3219                 21778         7552                             2219   
3220                 29868         2001                             5900   
3221                 33509        24597                              649   

      Hispanic or Latino  
0                   2188  
1                  13393  
2                   1490  
3                    744  
4                   5962  
...                  ...  
3217               53036  
3218                7803  
3219               21700  
3220               29732  
3221               33243  

[3222 rows x 12 columns]

In [947]:
print("Dataset Shape:", df.shape)
print("\nGeneral Information:")
df.info()

print("\nFirst five rows:")
display(df.head())

print("\nLast five rows:")
display(df.tail())

Dataset Shape: (3222, 12)

General Information:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3222 entries, 0 to 3221
Data columns (total 12 columns):
 #   Column                           Non-Null Count  Dtype 
---  ------                           --------------  ----- 
 0   County                           3222 non-null   object
 1   State                            3222 non-null   object
 2   State FIPS Code                  3222 non-null   int64 
 3   County FIPS Code                 3222 non-null   int64 
 4   FIPS                             3222 non-null   int64 
 5   Total Population                 3222 non-null   int64 
 6   Male Population                  3222 non-null   int64 
 7   Female Population                3222 non-null   int64 
 8   Total Race Responses             3222 non-null   int64 
 9   White Alone                      3222 non-null   int64 
 10  Black or African American Alone  3222 non-null   int64 
 11  Hispanic or Latino               3222 non-null 

County    State  State FIPS Code  County FIPS Code  FIPS  \
0  Autauga County  Alabama                1                 1  1001   
1  Baldwin County  Alabama                1                 3  1003   
2  Barbour County  Alabama                1                 5  1005   
3     Bibb County  Alabama                1                 7  1007   
4   Blount County  Alabama                1                 9  1009   

   Total Population  Male Population  Female Population  Total Race Responses  \
0             59285            28669              30616                 59285   
1            239945           117316             122629                239945   
2             24757            12906              11851                 24757   
3             22152            11824              10328                 22152   
4             59292            29934              29358                 59292   

   White Alone  Black or African American Alone  Hispanic or Latino  
0        43616                            11829                2188  
1       198721                            19144               13393  
2        10891                            11616                1490  
3        16634                             4587                 744  
4        53062                              747                5962


Last five rows:


County        State  State FIPS Code  County FIPS Code  \
3217  Vega Baja Municipio  Puerto Rico               72               145   
3218    Vieques Municipio  Puerto Rico               72               147   
3219   Villalba Municipio  Puerto Rico               72               149   
3220    Yabucoa Municipio  Puerto Rico               72               151   
3221      Yauco Municipio  Puerto Rico               72               153   

       FIPS  Total Population  Male Population  Female Population  \
3217  72145             54058            25765              28293   
3218  72147              8147             4178               3969   
3219  72149             21778            10510              11268   
3220  72151             29868            14381              15487   
3221  72153             33509            15920              17589   

      Total Race Responses  White Alone  Black or African American Alone  \
3217                 54058        13681                             2249   
3218                  8147         1028                              222   
3219                 21778         7552                             2219   
3220                 29868         2001                             5900   
3221                 33509        24597                              649   

      Hispanic or Latino  
3217               53036  
3218                7803  
3219               21700  
3220               29732  
3221               33243

<span style="color:red; font-size:16px"><em>The dataset contains detailed demographic information for every county and state in the United States. It is structured with 3,222 rows (one for each county) and 12 columns, capturing data related to population, race, and ethnicity. The dataset includes the following variables:
- **'County'**: The name of each county.
- **'State'**: The state to which the county belongs.
- **'State FIPS Code'**: The Federal Information Processing Standards (FIPS) code for each state.
- **'County FIPS Code'**: The FIPS code for each county within the state.
- **'FIPS'**: A unique code combining the state and county FIPS codes.
- **'Total Population'**: The total population of the county.
- **'Male Population'**: The male population of the county.
- **'Female Population'**: The female population of the county.
- **'Total Race Responses'**: The total number of responses regarding race in the county.
- **'White Alone'**: The population of the county identifying as White only.
- **'Black or African American Alone'**: The population of the county identifying as Black or African American only.
- **'Hispanic or Latino'**: The population of the county identifying as Hispanic or Latino.

***Data Summary***:
- **Shape**: (3222, 12), indicating 3,222 counties and 12 attributes.
- ***Data Types***:
    - 10 integer columns (e.g., 'State FIPS Code', 'County FIPS Code', 'Total Population', etc.).
    - 2 object columns ('County', 'State').
- ***Memory Usage: The dataset occupies approximately 302.2 KB of memory. </em></span>


<span style="color:green; font-size:18px"><strong>2. Identify the variable types (categorical, numerical, datetime, etc.)</strong></span>


In [950]:
categorical_vars = df.select_dtypes(include=['object']).columns.tolist()
numerical_vars = df.select_dtypes(include=['number']).columns.tolist()
datetime_vars = df.select_dtypes(include=['datetime']).columns.tolist()

print("\nCategorical Variables:", categorical_vars)
print("Numerical Variables:", numerical_vars)
print("Datetime Variables:", datetime_vars)


Categorical Variables: ['County', 'State']
Numerical Variables: ['State FIPS Code', 'County FIPS Code', 'FIPS', 'Total Population', 'Male Population', 'Female Population', 'Total Race Responses', 'White Alone', 'Black or African American Alone', 'Hispanic or Latino']
Datetime Variables: []


<span style="color:red; font-size:16px"><em>The code above identifies and categorizes the variables in the DataFrame based on their data types. The 'categorical_vars' list includes columns with object types (like 'County' and 'State'), while 'numerical_vars' contains columns with numeric data (such as 'Total Population' and 'Male Population'). The 'datetime_vars' list is empty, indicating that there are no datetime variables in this dataset. This categorization is essential for determining how to handle each variable in subsequent analysis, such as encoding categorical data or applying statistical methods to numerical variables.</em></span>


<span style="color:green; font-size:18px"><strong>1. Are there any missing values?</strong></span><br>
- How many missing values?
- Briefly describe how you plan to handle them.

In [953]:
missing_values = df.isnull().sum()
print("\nMissing Values:")
print(missing_values[missing_values > 0])


Missing Values:
Series([], dtype: int64)


<span style="color:red; font-size:16px"><em>As shown above, there are no missing values in the DataFrame, so I don't need to do anything. If there were missing values, I would drop columns with excessive missing data.</em></span>


---

## SECTION II: DATA CLEANING & INDEXING


<span style="color:green; font-size:18px"><strong>4. Check for and remove duplicates if applicable.</strong></span><br>

In [957]:
duplicates = df.duplicated().sum()
print(f"Number of duplicates: {duplicates}")

Number of duplicates: 0


<span style="color:red; font-size:16px"><em>As shown above, there are no duplicates in the DataFrame, so I don't need to do anything. Without duplicates, we can proceed with further analysis without the need for data cleaning steps related to redundancy.</em></span>


<span style="color:green; font-size:18px"><strong>5. Provide two examples of DataFrame filtering using logical conditions.</strong></span><br>

<span style="color:red; font-size:16px"><em>The first filtering example below shows where the 'Black or African American Alone' population is greater than 100,000. This filtering allows us to focus on regions with larger Black or African American populations, which can be useful for studying specific community dynamics, needs, or trends in these areas. By isolating these populations, we can better understand patterns and make informed decisions on policies or outreach programs targeting larger communities.</em></span>

In [961]:
filtered_black_pop = df[df['Black or African American Alone'] > 100000]
print("Counties with Black or African American Alone population greater than 100,000:")
print(filtered_black_pop)

Counties with Black or African American Alone population greater than 100,000:
                 County       State  State FIPS Code  County FIPS Code   FIPS  \
36     Jefferson County     Alabama                1                73   1073   
48        Mobile County     Alabama                1                97   1097   
50    Montgomery County     Alabama                1               101   1101   
104     Maricopa County     Arizona                4                13   4013   
171      Pulaski County    Arkansas                5               119   5119   
...                 ...         ...              ...               ...    ...   
2744     Tarrant County       Texas               48               439  48439   
2751      Travis County       Texas               48               453  48453   
2850     Fairfax County    Virginia               51                59  51059   
2971        King County  Washington               53                33  53033   
3089   Milwaukee County   Wisc

<span style="color:red; font-size:16px"><em>The second filtering example below shows where the 'County' belongs to a specific state, e.g., 'California'. This type of filtering helps isolate data based on a given state, enabling more targeted analysis, such as understanding demographic trends or population distributions within that specific region. By narrowing down the dataset to a particular state like California, we can focus on the counties within it to perform state-specific assessments or comparisons.</em></span>

In [963]:
filtered_california = df[df['State'] == 'California']  # Make sure your dataset has a 'State' column
print("Counties in California:")
print(filtered_california)

Counties in California:
                     County       State  State FIPS Code  County FIPS Code  \
187          Alameda County  California                6                 1   
188           Alpine County  California                6                 3   
189           Amador County  California                6                 5   
190            Butte County  California                6                 7   
191        Calaveras County  California                6                 9   
192           Colusa County  California                6                11   
193     Contra Costa County  California                6                13   
194        Del Norte County  California                6                15   
195        El Dorado County  California                6                17   
196           Fresno County  California                6                19   
197            Glenn County  California                6                21   
198         Humboldt County  California 

---

## SECTION III: DESCRIPTIVE STATISTICS


<span style="color:green; font-size:18px"><strong>6. Use.describe() on numerical columns. Share at least one interesting insight.</strong></span><br>

In [966]:
print("Descriptive Statistics for Numerical Columns:")
print(df.describe())

Descriptive Statistics for Numerical Columns:
       State FIPS Code  County FIPS Code          FIPS  Total Population  \
count      3222.000000       3222.000000   3222.000000      3.222000e+03   
mean         31.274674        103.216325  31377.890441      1.041721e+05   
std          16.285065        106.560811  16299.548258      3.296689e+05   
min           1.000000          1.000000   1001.000000      4.300000e+01   
25%          19.000000         35.000000  19029.500000      1.098550e+04   
50%          30.000000         79.000000  30022.000000      2.596700e+04   
75%          46.000000        133.000000  46104.500000      6.752625e+04   
max          72.000000        840.000000  72153.000000      9.848406e+06   

       Male Population  Female Population  Total Race Responses   White Alone  \
count     3.222000e+03       3.222000e+03          3.222000e+03  3.222000e+03   
mean      5.154751e+04       5.262457e+04          1.041721e+05  6.580439e+04   
std       1.628260e+05    

<span style="color:red; font-size:16px"><em>Despite the average county having about 104,000 people, the median is only around 25,967—indicating that a small number of highly populated counties are skewing the average upward. This shows a **right-skewed distribution** in total population across counties. Most counties in the U.S. are relatively small, but a few—like those with major cities—have populations in the millions, which pulls the mean much higher than the median.</em></span>


<span style="color:green; font-size:18px"><strong>7. Use .value_counts() to analyze at least two categorical columns.</strong></span><br>

In [969]:
print("\nTop 10 Most Common Counties:")
print(df['County'].value_counts().head(10))


Top 10 Most Common Counties:
County
Washington County    30
Jefferson County     25
Franklin County      24
Jackson County       23
Lincoln County       23
Madison County       19
Montgomery County    18
Clay County          18
Marion County        17
Monroe County        17
Name: count, dtype: int64


<span style="color:red; font-size:16px"><em>This output indicates that there are multiple counties with common names in the dataset. For example, Washington County appears 30 times, Jefferson County 25 times, and so on, highlighting that these names are widely used across different states.</em></span>

In [971]:
print("\nTop 10 States by County Count:")
print(df['State'].value_counts().head(10))


Top 10 States by County Count:
State
Texas             254
Georgia           159
Virginia          133
Kentucky          120
Missouri          115
Kansas            105
Illinois          102
North Carolina    100
Iowa               99
Tennessee          95
Name: count, dtype: int64


<span style="color:red; font-size:16px"><em>This output shows the top 10 states with the highest number of counties. Texas leads by a large margin with 254 counties, followed by Georgia with 159, and Virginia with 133. This indicates that some states have a much more granular administrative division than others.</em></span>

<span style="color:green; font-size:18px"><strong>9. Use .groupby() to compute summary statistics (e.g., mean or count) across
categories.</strong></span><br>

In [974]:
if 'State' in df.columns:
    print("\nAverage Black or African American Alone Population by State:")
    print(df.groupby('State')['Black or African American Alone'].mean().sort_values(ascending=False))



Average Black or African American Alone Population by State:
State
District of Columbia    290772.000000
Maryland                 76078.333333
Delaware                 73548.333333
New Jersey               57193.000000
Florida                  50205.507463
New York                 47209.806452
Connecticut              42750.333333
California               37471.431034
Massachusetts            34956.428571
South Carolina           28665.869565
Arizona                  22462.066667
Louisiana                22421.140625
North Carolina           21783.290000
Georgia                  21331.377358
Pennsylvania             20800.238806
Alabama                  19679.208955
Nevada                   17400.117647
Illinois                 17160.921569
Ohio                     16437.113636
Michigan                 16225.168675
Texas                    14276.129921
Mississippi              13302.158537
Rhode Island             12772.400000
Virginia                 12203.240602
Tennessee           

<span style="color:red; font-size:16px"><em>This output shows the average population of Black or African American individuals, specifically those counted as "Black or African American Alone," across different states. The District of Columbia has the highest average at 290,772, followed by Maryland with about 76,078. States like New Jersey, Florida, and New York also have relatively high averages. This suggests that certain states, particularly on the East Coast, have significant Black or African American populations, which could be important for demographic studies, policy-making, or targeted community programs. Lower averages in some states may indicate smaller Black or African American populations relative to other racial groups.</em></span>


In [976]:
if 'State' in df.columns:
    print("\nTotal Female Population by State:")
    print(df.groupby('State')['Female Population'].sum().sort_values(ascending=False))



Total Female Population by State:
State
California              19636903
Texas                   14850356
Florida                 11155261
New York                10169902
Pennsylvania             6585606
Illinois                 6422254
Ohio                     5970969
Georgia                  5540828
North Carolina           5406453
Michigan                 5069516
New Jersey               4708343
Virginia                 4379009
Washington               3842772
Arizona                  3639481
Massachusetts            3575630
Tennessee                3558032
Indiana                  3434741
Maryland                 3168659
Missouri                 3127491
Wisconsin                2941483
Colorado                 2868206
Minnesota                2851582
South Carolina           2675318
Alabama                  2600834
Louisiana                2358203
Kentucky                 2276855
Oregon                   2124865
Oklahoma                 2006574
Connecticut              1833231
Pu

<span style="color:red; font-size:16px"><em>This output displays the total female population by state, based on a grouped sum of the 'Female Population' column. California ranks highest with over 19.6 million women, followed by Texas with 14.8 million and Florida with 11.1 million. These figures reflect the overall population size of these states, which are among the most populous in the U.S. This information can be useful for understanding demographic distribution, planning gender-focused programs, or identifying states with large female populations for targeted policy or outreach efforts.</em></span>

if 'State' in df.columns and 'White Alone' in df.columns:
    df['Black_to_White_Ratio'] = df['Black or African American Alone'] / df['White Alone']
    print("\nAverage Black to White Ratio by State:")
    print(df.groupby('State')['Black_to_White_Ratio'].mean().sort_values(ascending=False))

<span style="color:red; font-size:16px"><em>This was an interesting filtering: the output displays the average Black-to-White population ratio by state. The ratio compares the population of Black or African American individuals to that of White individuals in each state. Mississippi has the highest ratio at 1.25, meaning there are more Black or African American individuals than White individuals on average in the state. The District of Columbia follows with a ratio of 1.11. On the other hand, states like Maine, Idaho, and Utah show much lower ratios, with minimal Black or African American populations relative to White populations. This ratio provides an interesting perspective on racial composition and could be useful for examining racial dynamics and social policies in different states.</em></span>


---

## SECTION IV: OBSERVATIONS AND QUESTIONS 


<span style="color:green; font-size:18px"><strong>9. Highlight 1-2 key takeaways or patterns from your descriptive analysis.</strong></span><br>

<span style="color:red; font-size:16px"><em>1. **Variation in 'Black or African American Alone' Population**: The average 'Black or African American Alone' population across states shows significant variation, with some states exhibiting a notably higher concentration of Black or African American residents, while others have much smaller numbers. For example, states like Mississippi and the District of Columbia have high ratios, while others like Wyoming and Montana have much lower numbers. This suggests a potential geographical pattern: states with higher Black populations might be more urbanized or have certain historical, economic, or social factors that contribute to a larger Black demographic. This could also indicate a concentration of Black communities in urban centers, which may offer more opportunities in terms of employment, education, and social networks. Further investigation into the demographic makeup of these urban counties and their historical context might reveal why these disparities exist.

<span style="color:red; font-size:16px"><em>2. **Discrepancies in Female and Male Population Data**: From the analysis of the 'Female Population' and 'Male Population' data, it's evident that there is often a higher female population in specific counties. This trend is seen across many states, and it may be linked to migration patterns or demographic factors that warrant deeper exploration. For instance, some counties may experience an influx of women due to opportunities in sectors like healthcare, education, or social services, where women are typically more represented. Alternatively, migration trends may play a role, as women may move to urban areas for better access to education and career opportunities. 

</span>


<span style="color:green; font-size:18px"><strong>9. Suggest 2 questions you would explore further in Part 2 using visualization.</strong></span><br>

<span style="color:red; font-size:16px"><em><strong>Question 1</strong>: How does the distribution of the Black or African American population vary across different counties and states?  
<span style="color:red; font-size:16px"><em><strong>Visualization idea</strong>: A heatmap or bar chart showing the concentration of the Black or African American population across states and counties. This would provide a visual representation of the distribution, helping identify states and counties with high or low Black populations. For example, we could highlight states like Mississippi or the District of Columbia that have a high proportion of Black or African American residents, compared to states with lower populations like Wyoming or Montana.


<span style="color:red; font-size:16px"><em><strong>Question 2</strong>: What is the relationship between the total population of a county and its racial demographics (e.g., Black, White)?  
<strong>Visualization idea</strong>: A scatter plot with population size on one axis and the ratio of Black to White population on the other, potentially color-coded by state or county. This would help visualize how population size correlates with racial demographic composition in different counties. Larger counties with diverse populations might show different trends compared to smaller or more homogenous counties. A color-coded plot could provide further insight into regional trends and disparities across states.</span>


---

CONCLUSION


<span style="color:red; font-size:16px">The analysis of county-level population data across the United States has uncovered important demographic patterns related to race, gender, and geographic distribution. The dataset reveals significant variation in the "Black or African American Alone" population across different states, with areas like the District of Columbia, Mississippi, and Maryland exhibiting high concentrations of Black residents, while states such as Wyoming, Montana, and Maine have much lower numbers. This geographical pattern likely reflects historical, social, and economic factors, with urbanized areas having a higher concentration of Black populations due to greater access to opportunities. Investigating the specific urban centers with high Black populations could provide more context on how factors like education, employment, and social networks influence these demographic trends.

<span style="color:red; font-size:16px">The dataset also highlights discrepancies between male and female populations, with a consistent trend of higher female populations in many counties. This imbalance could be driven by migration patterns where women are more likely to move to urban areas for opportunities in sectors such as healthcare, education, and social services, where they are traditionally more represented. Understanding these gender-based demographic shifts can help inform policies and initiatives aimed at supporting women in different regions, particularly in sectors where they are more prevalent.

<span style="color:red; font-size:16px">Another interesting finding from the analysis is the skewed distribution of county population sizes. The mean total population is much higher than the median, which indicates that a few large counties with major metropolitan areas are skewing the average upwards. This right-skewed distribution suggests that most counties in the U.S. are relatively small in population, while a few urban centers, such as those in California, Texas, and Florida, have significantly larger populations. These disparities are critical when considering resource allocation and policy development, as large urban centers may require different approaches compared to rural areas.

<span style="color:red; font-size:16px">The Black-to-White population ratio also stands out as a key insight. In states like Mississippi, the ratio is greater than 1, meaning that there are more Black or African American individuals than White individuals on average. This contrasts with states like Maine and Utah, where the ratio is much lower due to a smaller Black population. This racial composition metric provides a valuable lens for understanding regional dynamics and could inform social policies aimed at addressing racial disparities in different parts of the country.

<span style="color:red; font-size:16px">The analysis of categorical data, such as the frequency of county names and the number of counties by state, offers insight into the administrative structure of the U.S. Texas, for example, has a significantly higher number of counties compared to other states, which might influence how demographic data is reported and used for policymaking. The frequency of common county names like Washington and Jefferson across various states also reveals the complexity of demographic analysis and highlights the importance of understanding regional differences.

<span style="color:red; font-size:16px">Lastly, the total female population by state reflects trends consistent with overall population size, with California, Texas, and Florida having the largest female populations. These states' higher numbers are linked to their large total populations, while smaller states like Vermont and Wyoming show fewer females. Understanding these trends is crucial for designing gender-focused policies, particularly in larger states with diverse populations.

<span style="color:red; font-size:16px">In conclusion, the insights gained from this descriptive analysis open up numerous avenues for further exploration, particularly with the use of visualization techniques. For example, investigating how the Black or African American population varies across counties and states and examining the relationship between population size and racial demographics could provide deeper insights into the demographic landscape of the U.S. Overall, these findings contribute to a better understanding of the country's racial, gender, and geographic diversity, which is essential for developing targeted policies and initiatives.</strong></span><br>